Add simple ldap authentification with user/password and fix qa_get_user_email function

achorein opened the pull request:
Add server port configuration
Fixed qa_get_user_email function for email notification on answer or comment

Commit:
Add simple ldap authentification with user/password.
Add server port configuration
Fixed qa_get_user_email function for email notification on answer or comment

brenard commented:
Hi,

Thank you for this. I mean some details have to be fixed before merge :
- Auth have to be optionnal. I suggest that if constants _LDAP_SERVER_LOGIN_ (_LDAP_SERVER_BIND_DN_ could be a better name) and _LDAP_SERVER_PASSWORD_ (_LDAP_SERVER_BIND_PASSWORD_ so) are not defined, do not try to bind.
- _LDAP_PORT_ is not really necessary because _LDAP_SERVER_ constant could contain LDAP URI (Ex : _ldap://ldapserver:389_). Moreover, this constant permit to configure LDAPS connection using URI _ldaps://ldapserver:636_. I propose to rename _LDAP_SERVER_ constant in _LDAP_SERVER_URI_ and add comments for this possibilities.
- Finally, could you translate your errors messages in English.

Thank again

Commit:
- rename LDAP_SERVER_LOGIN to LDAP_SERVER_BIND_DN and LDAP_SERVER_PASSWORD to LDAP_SERVER_BIND_PASSWORD
- Make Auth optionnal, do not try to bind when LDAP_SERVER_BIND_DN is empty
- Replace LDAP_SERVER and LDAP_SERVER_PORT by LDAP_SERVER_URI
- Translate errors messages in English.

Commit:
get_ldap_userid_by_publicname : add ldap uri and bind

achorein commented:
jobs done and tested :)

brenard commented:
Hi,

Nice ! Thank you !
